src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -35,8 +36,18 @@ public Faculty removeFaculty(@PathVariable long id){
     public Collection<Faculty> getAllFaculty(){
         return facultyService.getAllFaculty();
     }
-    @GetMapping("/color")
-    public Collection<Faculty> searchForStudentsByColor(@RequestParam("color") String color){
-        return facultyService.searchForStudentsByColor(color);
+    @GetMapping("/")
+    public Collection<Faculty> searchForStudentsByColor(@RequestParam(value = "name", required = false) String name,
+                                                        @RequestParam(value = "color", required = false) String color){
+        return facultyService.findFacultyByNameOrColor(name, color);
     }
+    @GetMapping("/student{id}")
```

Comment:
```suggestion
    @GetMapping("/{id}/students")
```
логика - обращаемся к ресурсу факультет по id и берём у него студентов